src/vm/gcenv.ee.cpp

```diff
@@ -20,6 +20,8 @@ void GCToEEInterface::SuspendEE(SUSPEND_REASON reason)
 
     _ASSERTE(reason == SUSPEND_FOR_GC || reason == SUSPEND_FOR_GC_PREP);
 
+    g_pDebugInterface->BeforeGarbageCollectionTakesLock();
```

Comment:
It is not clear what is the lock that this talks about. Would `BeforeGarbageCollectionSuspectThread` make more sense?

Or rename these to `SuspendForGarbageCollectionStarted` and ``SuspendForGarbageCollectionFinished` to match what these events/callbacks are called in the profiler APIs?